app/bundles/LeadBundle/Entity/LeadRepository.php

```diff
@@ -218,28 +218,7 @@ function ($row) {
      */
     public function getLeadsByUniqueFields($uniqueFieldsWithData, $leadId = null, $limit = null)
     {
-        $q = $this->getEntityManager()->getConnection()->createQueryBuilder()
-            ->select('l.*')
-            ->from(MAUTIC_TABLE_PREFIX.'leads', 'l');
-
-        // loop through the fields and
-        foreach ($uniqueFieldsWithData as $col => $val) {
-            $q->{$this->getUniqueIdentifiersWherePart()}("l.$col = :".$col)
-                ->setParameter($col, $val);
-        }
```

Comment:
I had to add this loop back in https://github.com/mautic/mautic/pull/11697/commits/307a1177bd230da6985194e7800913eefa98d83d as it was failing many tests and Rector also complained. I think it was an oversight that it was removed and I don't understand why CI did not fail on this PR.

Reply:
@escopecz Hm? Code was added to a different method. Or i am missing something?

Reply:
It was suppose to be moved to the `getLeadFieldsByUniqueFields` method but notice that this loop is missing there and the `$uniqueFieldsWithData` param is unused which was flagged by Rector.

Reply:
The loop is in place in the PR. See https://github.com/mautic/mautic/pull/11716/files/e7d8f4eda92d54f64e2be8783e6f821873af5e6a#diff-579a524fc3ad315c33ef0c12e347b36ea85b86d258634444978c5ecbdb9d0b51R284

Reply:
I see. I had that part of code collapsed in the diff view. So it must have disappeared afterwards. Probably in some merge. Never mind, it's back in. Sorry for unnecessary confusioin.

Reply:
Lines were removed in 1446eb6210b59cd8904314d71aa90663b776bb69 as far as i can see.

Reply:
Yea, that must have been a conflict during rebase or cherry-pick as the code was moved around. My bad. Apologies.

Reply:
No problem, now we know why and can prevent it in the future. Thanks for your work!